chasm/field.go

```diff
@@ -26,29 +26,48 @@ package chasm
 
 import (
 	"reflect"
+	"strings"
 
 	"google.golang.org/protobuf/proto"
 )
 
+const (
+	// Used by reflection.
+	chasmFieldTypePrefix      = "chasm.Field["
+	chasmCollectionTypePrefix = "chasm.Collection["
+	internalFieldName         = "Internal"
+
+	fieldNameTag = "name"
+)
+
 // This struct needs to be create via reflection
 // but reflection can't set prviate fields...
 type Field[T any] struct {
 	Internal fieldInternal
```

Comment:
I wonder if we need to use `*fieldInternal`, in case user value is expensive to copy?
Probably not important right now. Component needs to be *struct and data is proto message, which is implement by pointer as well.

Reply:
Yes, user `value` is pointer to the struct and it will be enforced. I was thinking about it and decided to leave as value for now.